### PR TITLE
AUTH-134 fix(clerk-js,types): Resume an existing sign-up on SignUpStart only i…

### PR DESCRIPTION
**Pull request by yourtallness**
…f it's a ticket flow

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [X] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [X] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [X] `npm test` runs as expected.
- [X] `npm run build` runs as expected.

In the case of a ticket flow with missing requirements, the sign-up created for the ticket will be resumed when a social connection is initiated (since it may help complete the setup).

However this was done unconditionally in `authenticateWithRedirect()` and introduced an interaction with back button navigation from a verification screen back to the initial sign-up screen.

This PR explicitly instructs `authenticateWithRedirect()` to resume an existing sign-up if it's a ticket flow scenario, otherwise it will default to creating a new sign-up.

Note: Needs further manual testing & unit tests.

### CODE REVIEW

File: packages/clerk-js/src/core/resources/SignUp.ts [hunk 1]
```diff
@@ -203,7 +207,9 @@ export class SignUp extends BaseResource implements SignUpResource {
     });
   };
 
-  protected fromJSON(data: SignUpJSON | null): this {
+  protected;
+
+  fromJSON(data: SignUpJSON | null): this {
```
Comment (gkats) [inline packages/clerk-js/src/core/resources/SignUp.ts:212]: :wrench: Is this change intentional? I believe the `protected` modifier should come right before the method definition.

File: packages/types/src/oauth.ts [hunk 2]
```diff
@@ -244,4 +244,8 @@ export type AuthenticateWithRedirectParams = {
    * Full URL or path to navigate after the OAuth flow completes.
    */
   redirectUrlComplete: string;
+  /**
+   * Whether to continue (i.e. PATCH) an existing SignUp (if present) or create a new SignUp
+   */
+  continueSignUp?: boolean;
```
Comment (gkats) [inline packages/types/src/oauth.ts:250]: :upside_down_face: Not sure about property naming and whether this property should be part of the public API. 

How about `shouldCreateNew`? Let's see what the others think.

**Review by gkats — COMMENTED** [review]
Looks good to me overall!

File: packages/clerk-js/src/core/resources/SignUp.ts (hunk 1 above)
Comment (yourtallness) [reply packages/clerk-js/src/core/resources/SignUp.ts:212]: Messed up while auto-formatting, thanks for spotting

File: packages/clerk-js/src/ui/components/SignUp/SignUpSocialButtons.tsx [hunk 3]
```diff
@@ -10,7 +10,9 @@ import { SocialButtons } from '../../elements/SocialButtons';
 import { useNavigate } from '../../hooks';
 import { handleError } from '../../utils';
 
-export const SignUpSocialButtons = React.memo((props: SocialButtonsProps) => {
+export type SignUpSocialButtonsProps = SocialButtonsProps & { continueSignUp?: boolean };
```
Comment (dimkl) [inline packages/clerk-js/src/ui/components/SignUp/SignUpSocialButtons.tsx:13]: ❓ why didn't we add `continueSignUp?: boolean` to `SocialButtonsProps` type in `elements/SocialButtons` ?

**Review by dimkl — APPROVED** [review]

File: packages/clerk-js/src/ui/components/SignUp/SignUpSocialButtons.tsx (hunk 3 above)
Comment (yourtallness) [reply packages/clerk-js/src/ui/components/SignUp/SignUpSocialButtons.tsx:13]: Because it doesn't apply to sign-in.

File: packages/types/src/oauth.ts (hunk 2 above)
Comment (yourtallness) [reply packages/types/src/oauth.ts:250]: Since `shouldCreateNew` should be true by default, I'd prefer to go for a flag that should be false by default.

We have the concept of "continuing a sign-up" already, so it's not a new term.

Will proceed as is to unblock this fix, but open to revisiting this.
